pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -1382,7 +1382,8 @@ func TestSelectBestCandidate(t *testing.T) {
 			candidates, _, _ := pe.DryRunPreemption(ctx, tt.pod, nodeInfos, nil, offset, numCandidates)
 			s := pe.SelectCandidate(candidates)
 			if s == nil || len(s.Name()) == 0 {
-				return
+				// Currently, all test cases should get non-empty candidate from SelectCandidate.
```

Comment:
I would remove this comment.

It's not about the test cases, but the behavior of the function. Unless it's not?

Reply:
Why not cover the scenario here? It is possible no candidate returned, like all the allocated pods are with higher priorities. no?

Reply:
there is no scenario that is worth being tested and returns no candidate. So, let's just remove the comment as @alculquicondor suggested.

> It is possible no candidate returned, like all the allocated pods are with higher priorities. no?

No. (Actually, I also misunderstood like you 😓  when I created #115001)
Before `SelectCandidate`, `findCandidates` is filtering out those kinda Nodes which cannot be target of preemption.
https://github.com/kubernetes/kubernetes/blob/a66aad2d80dacc70025f95a8f97d2549ebd3208c/pkg/scheduler/framework/preemption/preemption.go#L212
And `SelectCandidate` is just deciding which Node among them is the best while checking PDB etc ([ref](https://github.com/kubernetes/kubernetes/blob/a66aad2d80dacc70025f95a8f97d2549ebd3208c/pkg/scheduler/framework/preemption/preemption.go#L432-L437)).

Reply:
what's the latest decision?

Reply:
Have we came to an agreement https://github.com/kubernetes/kubernetes/pull/115001#discussion_r1070218635 @sanposhiho